### ISSUE TITLE
Support createIndex with specifying index prefix length

### ISSUE THREAD

**Issue by vaishakhikulkarni** [liquibase/liquibase-percona#148]
In SQL:

ALTER TABLE some_table MODIFY `column1` varchar(4096) CHARACTER SET utf8mb4 NOT NULL;
ALTER TABLE  some_table ADD INDEX `some_table_index ` (column1 (300),column2);

In Liquibase also we don't support override but have work around like follows:
First createIndex and use modifySql to replace column1 with column1(300) override size.

```xml
    <createIndex tableName="some_table" indexName="some_table_index">
        <column name="column1"/>
        <column name="column2"/>
    </createIndex>
    <modifySql dbms="mysql">
        <replace replace="`column1 `" with="`column1 `(300)"/>  <!-- "300" limits index size -->
    </modifySql>
```

**While we do not have any support or work around for liquibase percona. This would be very useful feature for us.**

**Comment by adangel** [comment]
Do I understand you correctly: You basically want that `modifySql` is also applied when the `ADD INDEX` statement is executed via percona? I guess, currently the whole "modifySql" is simply ignored...

Just some docs:
* https://docs.liquibase.com/change-types/community/create-index.html
* https://github.com/liquibase/liquibase-percona#createindex
* https://docs.liquibase.com/workflows/liquibase-community/modify-sql.html

I guess, this should then be supported not only for createIndex but basically for all change types.

**Comment by vaishakhikulkarni** [comment]
No I don't want modifySql to be applied for add Index.
I want a way to override the size of the column while creating a index using createIndex tag as we can do using SQL. 
ALTER TABLE some_table MODIFY column1 varchar(4096) CHARACTER SET utf8mb4 NOT NULL;
ALTER TABLE some_table ADD INDEX some_table_index (column1 (300),column2);

In liquibase also we don't have way to override the size of the column while creating index for tag createIndex.
Work around used is to add modifySQL tag.

Because even if we mention type while creating index it gets ignored in liquibase as well as liquibase percona.
<createIndex tableName="some_table" indexName="some_table_index">
        <column name="column1" type="varchar(300)"/>
        <column name="column2"/>
    </createIndex>

Please let me know accordingly.

**Comment by adangel** [comment]
I've created a fix for this, see #157.
You can download the fixed liquibase-percona extension here: 
[liquibase-percona-4.6.2-SNAPSHOT-fixed-148.zip](https://github.com/liquibase/liquibase-percona/files/7490732/liquibase-percona-4.6.2-SNAPSHOT-fixed-148.zip)

Would you mind testing this fix?

**Comment by mahameed** [comment]
@adangel My colleague @vaishakhikulkarni is out of office these days. I tested your update and it does indeed work now. However, our problem of not be able to change column size while updating index still remains. It seems like the 2 solutions to them would be to:

- Ask the liquibase team to update the size of the column in a `createindex` tag instead of having the `modifySql`
- Percona starts supporting `modifySql` tag, especially for createindex

Do you think the 2nd option is something viable?

**Comment by adangel** [comment]
@mahameed Thanks for testing.

Regarding your follow-up question: You can specify the column size while *creating* the index like this:

```xml
<createIndex tableName="some_table" indexName="some_table_index">
        <column name="column1(300)"/>
        <column name="column2"/>
</createIndex>
```

Is there anything else, what should be supported by liquibase? If so, please comment on https://github.com/liquibase/liquibase/issues/2191

So I don't think, modifySql is required at all (at least not for this use case).

**Comment by mahameed** [comment]
You can specify the column like you suggest but it's simply ignored by liquibase. Therefore it does not modify the size at all.

**Comment by mahameed** [comment]
@adangel ok I missed that part. Yeah that seems to be working. Thanks.
Any rough estimate on when 4.6.2 proper will be released?

**Comment by adangel** [comment]
I've released now 4.6.1.1: https://github.com/liquibase/liquibase-percona/releases/tag/liquibase-percona-4.6.1.1
